exchanges/kraken/kraken_websocket.go

```diff
@@ -1105,211 +924,365 @@ func trim(s string) string {
-				err = k.Websocket.AddSuccessfulSubscriptions((*subs)[i].Channels...)
-			}
-			if err != nil {
-				errs = common.AppendError(errs, err)
-			}
-		}
+	for err, pairs := range errPairs {
+		errs = common.AppendError(errs, fmt.Errorf("%w; Channel: %s Pairs: %s", err, s.Channel, pairs.Join()))
 	}
+
+	if k.Verbose && len(subPairs) > 0 {
+		log.Debugf(log.ExchangeSys, "%s Subscribed to Channel: %s Pairs: %s", k.Name, s.Channel, subPairs.Join())
+	}
+
 	return errs
 }
 
-// Unsubscribe sends a websocket message to stop receiving data from the channel
-func (k *Kraken) Unsubscribe(channelsToUnsubscribe subscription.List) error {
-	var unsubs []WebsocketSubscriptionEventRequest
-channels:
-	for x := range channelsToUnsubscribe {
-		for y := range unsubs {
-			if unsubs[y].Subscription.Name == channelsToUnsubscribe[x].Channel {
-				unsubs[y].Pairs = append(unsubs[y].Pairs, channelsToUnsubscribe[x].Pairs.Strings()...)
-				unsubs[y].Channels = append(unsubs[y].Channels, channelsToUnsubscribe[x])
-				continue channels
-			}
-		}
-		var depth int64
-		if channelsToUnsubscribe[x].Channel == "book" {
-			depth = krakenWsOrderbookDepth
-		}
+// getSubErrResp calls getRespErr and if there's no error from that ensures the status matches the sub operation
+func (k *Kraken) getSubRespErr(resp []byte, op string) error {
+	if err := k.getRespErr(resp); err != nil {
+		return err
+	}
+	exp := op + "d" // subscribed or unsubscribed
+	if status, err := jsonparser.GetUnsafeString(resp, "status"); err != nil {
+		return fmt.Errorf("error parsing WS status: %w from message: %s", err, resp)
+	} else if status != exp {
+		return fmt.Errorf("wrong WS status: %s; expected: %s from message %s", exp, op, resp)
+	}
 
-		var id int64
-		if common.StringDataContains(authenticatedChannels, channelsToUnsubscribe[x].Channel) {
-			id = k.Websocket.AuthConn.GenerateMessageID(false)
-		} else {
-			id = k.Websocket.Conn.GenerateMessageID(false)
-		}
+	return nil
+}
 
-		unsub := WebsocketSubscriptionEventRequest{
-			Event: krakenWsUnsubscribe,
-			Pairs: []string{channelsToUnsubscribe[x].Pairs[0].String()},
-			Subscription: WebsocketSubscriptionData{
-				Name:  channelsToUnsubscribe[x].Channel,
-				Depth: depth,
-			},
-			RequestID: id,
-		}
-		if common.StringDataContains(authenticatedChannels, channelsToUnsubscribe[x].Channel) {
-			unsub.Subscription.Token = authToken
+// getRespErr takes a json response string and looks for an error event type
+// If found it returns the errorMessage
+// It might log parsing errors about the nature of the error
+// If the error message is not defined it will return a wrapped errUnknownError
+func (k *Kraken) getRespErr(resp []byte) error {
+	event, err := jsonparser.GetUnsafeString(resp, "event")
+	switch {
+	case err != nil:
+		return fmt.Errorf("error parsing WS event: %w from message: %s", err, resp)
+	case event != "error":
+		status, _ := jsonparser.GetUnsafeString(resp, "status") // Error is really irrellevant here
```

Comment:
```suggestion
		status, _ := jsonparser.GetUnsafeString(resp, "status") // Error is really irrelevant here
```
Resolve if actual spelling and my checker is being all American

Reply:
Nope, that one is on me. Soz.

Fixed 2813a9fc